### ISSUE TITLE
Generator Gradient tape not calculating gradients

### ISSUE THREAD

**Issue by NiloyPurkait** [dbpedia/RDF2text-GAN#1]
### Update

_**The issue persists, despite several updates to the code.**_

- Latest code can be viewed in [this](https://github.com/dbpedia/RDF2text-GAN/blob/master/Transformers/Adversarial_Training_latest.ipynb) notebook, with the name:  Adversarial_Training_latest.ipynb .
- [This blog post](https://medium.com/@niloypurkait/when-gradients-just-wont-flow-b30c79182d1e) is one I made to describe the issue and the background settings in detail.

### Previous post

**Issue**

The '.gradient()' method on the generator's gradient tape returns None values, instead of calculating the gradients of the model.
This error only occurs during adversarial training. I am able to pre-train the generator without any problems (see generator_pretraining.ipynb' 

**I suspect that this issue is related to the ones discussed below:**

-  [similar issue from tensorflow page](https://github.com/tensorflow/tensorflow/issues/30190), 
- [another similar issue from tensorflow page](https://github.com/tensorflow/tensorflow/issues/29942) 
- [stackoverflow question with same error](https://stackoverflow.com/questions/51994043/tf-gradienttape-return-none). 
- [blogpost about possible problems with gradient tape](http://blog.ai.ovgu.de/posts/jens/2019/001_tf20_pitfalls/index.html)

**What might be going on here?**
My impression is that the operations performed on the generator's output (i.e. converting the sub-word encoding outputs to text, striping the generations by the end character, to remove all characters after the end of sequence token, then concatenating the input sequence with the generated output to feed to the discriminator) is breaking the tensorflow ops graph of the generator, thereby returning None values for the gradients of the model. The discriminator's tape on the other hand does indeed register gradients- it is only the generator's tape that does not seem to do so.

**Solutions tried**
I have tried many solutions, including watching variables with 'tape.watch()' syntax, and wrapping all functions used during training into tensorflow ops by using the 'tf.py_function()' syntax.  I have also tried coding the training step differently, using nested gradient tapes and sequential gradient tapes.

**Comment by NiloyPurkait** [comment]
The issue persists, despite several updates to the code.

Latest code can be viewed in [this](https://github.com/dbpedia/RDF2text-GAN/blob/master/Transformers/Adversarial_Training_latest.ipynb) notebook, with the name:  Adversarial_Training_latest.ipynb .

[This blog post](https://medium.com/@niloypurkait/when-gradients-just-wont-flow-b30c79182d1e) is one I made to describe the issue and the background settings in detail.

**Comment by zwingaa** [comment]
Hello, did you finally solve this problem, or replace it with Pytorch
